node/core/pvf/worker/src/execute.rs

```diff
@@ -94,26 +94,37 @@ pub fn worker_entrypoint(socket_path: &str, node_version: Option<&str>) {
 			let cpu_time_start = ProcessTime::now();
 
 			// Spawn a new thread that runs the CPU time monitor.
-			let cpu_time_monitor_fut = rt_handle
-				.spawn_blocking(move || {
-					cpu_time_monitor_loop(cpu_time_start, execution_timeout, finished_rx)
-				})
-				.fuse();
+			let cpu_time_monitor_thread = thread::spawn(move || {
+				cpu_time_monitor_loop(cpu_time_start, execution_timeout, finished_rx)
+			});
 			let executor_2 = executor.clone();
-			let execute_fut = rt_handle
-				.spawn_blocking(move || {
+			let execute_thread =
+				thread::Builder::new().stack_size(EXECUTE_THREAD_STACK_SIZE).spawn(move || {
 					validate_using_artifact(&artifact_path, &params, executor_2, cpu_time_start)
-				})
-				.fuse();
-
-			pin_mut!(cpu_time_monitor_fut);
-			pin_mut!(execute_fut);
-
-			let response = select_biased! {
-				// If this future is not selected, the join handle is dropped and the thread will
+				})?;
+
+			// "Select" the first thread that finishes by checking all threads in a naive loop.
+			let response = loop {
+				// NOTE: The order in which we poll threads is important! This loop sleeps between
```

Comment:
This should hint that mimicking async code is not the way for managing threads, (almost) busy-waiting is not nice either.

Any reason not to use condition variable?

Reply:
> Any reason not to use condition variable?

I forgot about them. 😅 Done, left some notes on design decision in comments and commit message.